File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/events/factory/V8Engine.kt
package io.micronaut.docs.events.factory

// tag::class[]
class V8Engine(var rodLength: Double) : Engine {  // <1>

    override val cylinders = 8

    override fun start(): String {
        return "Starting V$cylinders [rodLength=$rodLength]"
    }
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/docs/factories/V8Engine.py
from .CrankShaft import CrankShaft
from .Engine import Engine

# tag::class[]
class V8Engine(Engine):
    def __init__(self, crank_shaft: CrankShaft):
        self.crank_shaft = crank_shaft
        self.cylinders = 8

    def start(self) -> str:
        return "Starting V8"
# end::class[]
